src/Engine/Network/UDPClient.cpp

```diff
@@ -72,6 +84,72 @@ int UDPClient::readBuffer()
     return bytesReceived;
 }
 
+void UDPClient::readPartOfPacket()
+{
+    if (!m_Socket) {
+        return;
+    }
+    boost::system::error_code error;
+    // Peek header
+    m_Socket->receive(boost
+        ::asio::buffer((void*)m_ReadBuffer, 5 * sizeof(int)),
+        boost::asio::ip::udp::socket::message_peek, error);
+
+    int sizeOfPacket = 0;
+    memcpy(&sizeOfPacket, m_ReadBuffer, sizeof(int));
+    if (sizeOfPacket == 0) {
+        return;
+    }
+    int packetGroup = 0;
+    memcpy(&packetGroup, m_ReadBuffer + sizeof(int), sizeof(int));
```

Comment:
We could do `static_cast` here, but perhaps memcpy actually gives better performance. Not sure.